client/components/cards/ItemUploadCard.vue

```diff
@@ -105,9 +128,42 @@ export default {
     titleUpdated() {
       this.error = ''
     },
+    async fetchMetadata() {
+      if (!this.itemData.title.trim().length) {
+        return
+      }
+
+      this.isFetchingMetadata = true
+      this.error = ''
+
+      try {
+        const searchQueryString = this.buildQuerystring({
+          title: this.itemData.title,
+          author: this.itemData.author,
+          provider: this.provider
+        })
```

Comment:
```suggestion
        const searchQueryString = new URLSearchParams({
          title: this.itemData.title,
          author: this.itemData.author,
          provider: this.provider
        })
```

I don't think we need the mixin since the same thing can be achieved with URLSearchParams. It encodes the strings properly. I may be missing something though, let me know what you think.

Reply:
totally forgot about that guy! thanks for the tip 🤙